### PR TITLE
Fix two documentation bits regarding HTTP pipelining

### PR DESCRIPTION
**Pull request by jrudolph**
/cc @sirthias

### CODE REVIEW

**Comment by ktoso** [comment]
Refs #16491 #17595

File: akka-docs/rst/java/http/server-side/low-level-server-side-api.rst [hunk 1]
```diff
@@ -171,6 +171,33 @@ You create an instance of the layer by calling one of the two overloads of the `
...
+relying on HTTP pipelining to send several requests on one connection without waiting for a response first. In both
+cases the client controls the number of ongoing requests. To prevent being overloaded by too many requests, Akka HTTP
+can limit the number of requests it handles in parallel.
+
+To limit the number of simultaneously open connections, use the ``akka.http.server.max-connections`` setting. This setting
+applies to all of ``Http.bindAndHandle*`` methods. If you use ``Http.bind``, incoming connections are represented by
+a ``Source<IncomingConnection, ...>``. Use Akka Stream's combinators to apply backpressure to control the flow of
+incoming connections, e.g. by using ``throttle`` or ``mapAsync``.
+
+HTTP pipelining is generally discouraged (and [disabled by most
```
Comment (ktoso) [inline akka-docs/rst/java/http/server-side/low-level-server-side-api.rst]: Sorry about `rst` :wink: 
Syntax is 

```
`Blabla <https://avro.apache.org>`_
```

Comment (jrudolph) [inline akka-docs/rst/java/http/server-side/low-level-server-side-api.rst]: Haha, thanks, you always have to stay alert when editing documentation.

File: akka-http-core/src/main/resources/reference.conf [hunk 2]
```diff
@@ -198,6 +198,13 @@ akka.http {
     # This value must be > 0.
     # (Note that, independently of this setting, pipelining will never be done
     # on a connection that still has a non-idempotent request in flight.
+    #
+    # Before increasing this value, make sure you understand the effects of head-of-line blocking.
+    # Using a connection pool, a request may be issued on a connection where a previous
+    # long-running request hasn't finished yet. The response to the pipelined requests may then be stuck
+    # behind the response of the long-running previous requests on the server. This may introduce an
+    # unwanted "coupling" of run time between otherwise unrelated requests.
+    #
```
Comment (ktoso) [inline akka-http-core/src/main/resources/reference.conf]: very good warning

**Comment by ktoso** [comment]
LGTM, very nice, thanks @jrudolph !

**Comment by jrudolph** [comment]
Fixed the link.

**Comment by ktoso** [comment]
Thanks :)

**Comment by johanandren** [comment]
LGTM!

**Pull request merged**